### ISSUE TITLE
number type allows NaN but `${number}` template literal type does not allow "NaN"

### ISSUE THREAD

**Issue by DetachHead** [microsoft/TypeScript#42996]
# Bug Report

### 🔎 Search Terms

NaN template literal types
### 🕗 Version & Regression Information
4.3.0-dev.20210226
### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210226#code/MYewdgzgLgBAZiEAuGYCuBbARgUwE4wC8MAcgIYkwD0VYIM+eIeAsAFCiSxZl4oAGAEgDe6bPgC+-IjADk5ErOpUAKgE8ADjjkAiBTqUBLCKhCwyECIYDmYMlgA22qPSibtsoaMy48U2QB0QA)

### 💻 Code

```ts
const foo: number = NaN //no error
const bar: `${number}` = 'NaN' //Type '"NaN"' is not assignable to type '`${number}`'.
```

### 🙁 Actual behavior

``Type '"NaN"' is not assignable to type '`${number}`'.`` error
### 🙂 Expected behavior

the string `"NaN"` should be a valid `` `${number}` `` because `NaN` is a valid `number`

**Comment by RyanCavanaugh** [comment]
I'm not really sure this is what's intended 99.9% of the time.

**Comment by DetachHead** [comment]
Me neither. I recently added template literal types to the typescript-is runtime type checker and wanted to make sure I made it behave the same way the typescript compiler does https://github.com/woutervh-/typescript-is/pull/92

**Comment by gabrielurbina** [comment]
Maybe useful detail or workarround, this works, casting it into number works.

```typescript
const b: `${number}` = `${+NaN}`;
const d: `${number}` = `${-Infinity}`;
```

**Comment by rotu** [comment]
> I'm not really sure this is what's intended 99.9% of the time.

No, but almost all the time, what's intended is actually a non-negative integer. It's just that TypeScript doesn't have such a type.

`-0` is already allowed. I would definitely expect `Infinity`, `-Infinity`, `NaN` to also be admissible values of `${number}`.
